pytest_html/plugin.py

```diff
@@ -127,18 +131,36 @@ def __lt__(self, other):
                      'XPassed', 'Skipped', 'Passed')
             return order.index(self.outcome) < order.index(other.outcome)
 
-        def append_extra_html(self, extra, additional_html, links_html):
+        def append_extra_html(self, extra):
             href = None
             if extra.get('format') == extras.FORMAT_IMAGE:
-                href = '#'
-                image = 'data:image/png;base64,{0}'.format(
-                        extra.get('content'))
-                additional_html.append(html.div(
-                    html.a(html.img(src=image), href="#"),
+                image_src = None
+                if self.contained_html:
+                    image_src = 'data:image/png;base64,{0}'.format(
+                            extra.get('content'))
+                    href = '#'
+                else:
+                    hash_key = self.test_id + self.outcome
```

Comment:
The only thing that the `outcome` may help is with rerun, because the the test which failed at first may have a different picture, but  I don't see a way to get the index of the rerun. Do you have any idea?